### PR TITLE
220105 network tc refactoring

### PR DESCRIPTION
**Pull request by pillip8282**

### CODE REVIEW

File: apps/examples/testcase/le_tc/network/tc_net_shutdown.c [hunk 1]
```diff
@@ -278,23 +274,21 @@ void *shutdown_client(void *args)
 		printf("\nShutdown recv successful %d\n", errno);
 	tc_net_shutdown_n();
 	close(mysocket);
-	return 0;
 
+	return 0;
```
[the remark this answers is not in the document]
Comment (pillip8282) [reply apps/examples/testcase/le_tc/network/tc_net_shutdown.c:278]: Thanks for review. I'll modify it

File: apps/examples/testcase/le_tc/network/itc_net_connect.c [hunk 2]
```diff
@@ -35,6 +36,57 @@
...
+static sem_t g_sig_clitomain;
+static sem_t g_sig_srvtomain;
+
+static int _recv(int sd, char *buf, int buf_size)
+{
+	int recved = 0;
+	while (recved < buf_size) {
+		int ret = recv(sd, buf + recved, buf_size - recved, MSG_WAITALL);
+		if (ret <= 0) {
+			printf("recv error %d %s %d %s:%d\n",ret,  strerror(errno), __FUNCTION__, __LINE__);
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/network/itc_net_connect.c:48]: There are five formats `%d %s %d %s:%d`, but there are only four variables.
Please check them.
And please add space after `,` like below.
```
\n", ret, ...
```

File: apps/examples/testcase/le_tc/network/itc_net_connect.c [hunk 3]
```diff
@@ -35,6 +36,57 @@
...
+	return 0;
+}
+
+static int _send(int sd, char *buf, int buf_size)
+{
+	int sent = 0;
+	while (sent < buf_size) {
+		int ret = send(sd, buf + sent, buf_size - sent, 0);
+		if (ret <= 0) {
+			printf("sent error %d %s %d %s:%d\n", ret, strerror(errno), __FUNCTION__, __LINE__);
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/network/itc_net_connect.c:62]: Same as above. Please check the params.

File: apps/examples/testcase/le_tc/network/itc_net_connect.c [hunk 4]
```diff
@@ -304,40 +315,52 @@ static void itc_net_connect_p_multiple_clients(void)
 	int id2 = 2;
 	int id3 = 3;
 	int ret;
-	int *ptr_num_clients = &num_clients;
-	int *ptr_id1 = &id1;
-	int *ptr_id2 = &id2;
-	int *ptr_id3 = &id3;
 	int *pret = OK;
 
+	_initialize_test();
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/network/itc_net_connect.c:320]: Is it okay not to check the initialization?
In other tc, you checked and return if failed.

File: apps/examples/testcase/le_tc/network/itc_net_connect.c (hunk 2 above)
Comment (pillip8282) [reply apps/examples/testcase/le_tc/network/itc_net_connect.c:48]: Thanks I'll modify it

File: apps/examples/testcase/le_tc/network/itc_net_connect.c (hunk 3 above)
Comment (pillip8282) [reply apps/examples/testcase/le_tc/network/itc_net_connect.c:62]: Thanks I'll modify it

File: apps/examples/testcase/le_tc/network/itc_net_connect.c (hunk 4 above)
Comment (pillip8282) [reply apps/examples/testcase/le_tc/network/itc_net_connect.c:320]: Thanks I'll modify it

File: apps/examples/testcase/le_tc/network/itc_net_connect.c [hunk 5]
```diff
@@ -263,26 +264,26 @@ static void itc_net_connect_p(void)
...
 
-	sleep(CLIENT_WAIT_TIME);
+	pthread_create(&server_thread, NULL, server_connect, (void *)&num_clients);
+	sem_wait(&g_sig_srvtomain);
+	pthread_create(&client_thread, NULL, client_connect, (void *)&id);
 
 	ret = pthread_join(server_thread, (void *)&pret);
 	TC_ASSERT_EQ_CLEANUP("pthread_join", ret, OK, free(pret); pthread_join(client_thread, NULL));
 	TC_ASSERT_EQ_CLEANUP("pthread_join", *pret, OK, free(pret); pthread_join(client_thread, NULL));
```
Comment (an4967) [inline apps/examples/testcase/le_tc/network/itc_net_connect.c:278]: Is it okay to use '*pret' even pthread returns NULL value?

Comment (pillip8282) [reply apps/examples/testcase/le_tc/network/itc_net_connect.c:278]: You're right. it's possible to free NULL if server_thread return NULL. It seems difficult problem. So I'll modify it next PR. Thanks for your analysis.

**Review by an4967 — APPROVED** [review]
LGTM

File: apps/examples/testcase/le_tc/network/itc_net_connect.c (hunk 5 above)
Comment (an4967) [reply apps/examples/testcase/le_tc/network/itc_net_connect.c:278]: Okay, I see

**Pull request merged**